vod/hls/m3u8_builder.c

```diff
@@ -398,11 +400,33 @@ m3u8_builder_build_index_playlist(
 		return VOD_ALLOC_FAILED;
 	}
 
+	// find the max segment duration
+	max_segment_duration = 0;
+	for (cur_item = segment_durations.items; cur_item < last_item; cur_item++)
+	{
+		if (cur_item->duration > max_segment_duration)
+		{
+			max_segment_duration = cur_item->duration;
+		}
+	}
+
+	// Note: scaling first to 'scale' so that target duration will always be round(max(manifest durations))
+	scale = conf->m3u8_version >= 3 ? 1000 : 1;
+	max_segment_duration = rescale_time(max_segment_duration, segment_durations.timescale, scale);
+	max_segment_duration = rescale_time(max_segment_duration, scale, 1);
+
+	// make sure segment duration is not lower than the value set in the conf
+	conf_max_segment_duration = (segmenter_conf->max_segment_duration + 500) / 1000;
```

Comment:
You probably meant round... anyway, yes there are functions like that in math.h, but I don't use it, since:
1. they work with floats, and floats are slow. I don't use floats unless I absolutely have to (for example, in HDS, I need to output a metadata AMF that contains some float value)
2. this doesn't justify bringing in another library
3. for such a simple logic, this is actually clearer to me